src/gui/editorwidgets/qgsvaluemapconfigdlg.cpp

```diff
@@ -175,16 +199,58 @@ void QgsValueMapConfigDlg::updateMap( const QList<QPair<QString, QVariant>> &lis
     ++row;
   }
 
+  constexpr int maxOverflowErrors { 5 };
+  QStringList reportedErrors;
+  const QgsField mappedField { layer()->fields().field( field() ) };
+
   for ( const auto &pair : list )
   {
     if ( QgsVariantUtils::isNull( pair.second ) )
       setRow( row, pair.first, QString() );
     else
-      setRow( row, pair.first, pair.second.toString() );
+    {
+      const QString value { pair.first };
+      // Check value
+      const QString validValue = checkValueLength( value );
+
+      if ( validValue.length() != value.length() )
+      {
+
+        if ( reportedErrors.length() < maxOverflowErrors )
+        {
+          reportedErrors.push_back( tr( "Value '%1' has been trimmed (maximum field length: %2)" )
+                                    .arg( value, QString::number( mappedField.length() ) ) );
+        }
+        else if ( reportedErrors.length() == maxOverflowErrors )
+        {
+          reportedErrors.push_back( tr( "Only first %1 errors have been reported." )
+                                    .arg( maxOverflowErrors ) );
+        }
+      }
+
+      setRow( row, validValue, pair.second.toString() );
+
+      // Show errors if any
+      if ( !reportedErrors.isEmpty() )
+      {
+        mValueMapErrorsLabel->setVisible( true );
+        mValueMapErrorsLabel->setText( reportedErrors.join( QStringLiteral( "<br>" ) ) );
+      }
+    }
     ++row;
   }
 }
 
+QString QgsValueMapConfigDlg::checkValueLength( const QString &value )
+{
+  const QgsField mappedField { layer()->fields().field( field() ) };
```

Comment:
Can we protect from fields() empty or field() equals to -1, even though is not suppose to happen

Reply:
done